controllers/awscluster_controller.go

```diff
@@ -62,6 +64,7 @@ type AWSClusterReconciler struct {
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusters/status,verbs=get;update;patch
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=clusters;clusters/status,verbs=get;list;watch
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awsclusterroleprincipals;awsclusterstaticprincipals;awsclustercontrollerprincipals,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Does the controller need all the verbs listed here? Would `get;update;pathc` be sufficient?

Reply:
I agree. Let's separate out awsclustercontrollerprincipals onto it's on line with the create permission